connector/connector_ldap.go

```diff
@@ -217,11 +238,74 @@ func (c *LDAPConnector) TrustedEmailProvider() bool {
 	return c.trustedEmailProvider
 }
 
+type LDAPPool struct {
+	m          sync.Mutex
+	conns      map[*ldap.Conn]*ldap.Conn
+	initConn   int
+	maxConn    int
+	serverHost string
+	serverPort uint16
+	useTLS     bool
+	useSSL     bool
+	tlsConfig  *tls.Config
+}
+
+func (p *LDAPPool) fillPool() error {
+	if p.conns == nil {
+		p.conns = make(map[*ldap.Conn]*ldap.Conn)
+	}
+	for i := len(p.conns); i < p.initConn; i++ {
+		ldapConn, err := p.ldapConnect()
+		if err != nil {
+			return err
+		}
+		p.conns[ldapConn] = ldapConn
+	}
+	return nil
+}
+
+func (p *LDAPPool) Get() (*ldap.Conn, error) {
+	p.m.Lock()
+	defer p.m.Unlock()
+	if p.conns == nil {
+		err := p.fillPool()
+		if err != nil {
+			return nil, err
+		}
+	}
+	for conn := range p.conns {
+		delete(p.conns, conn)
+		return conn, nil
+	}
+	// the pool is empty
+	conn, err := p.ldapConnect()
```

Comment:
Go's net/http.Transport has some pretty neat tricks for [connection caching](https://github.com/golang/go/blob/7c22af830a72adf30b18ee3d1744aab8e3c009a8/src/net/http/transport.go#L668) if you're looking for inspiration. Though this definitely doesn't need to be a sophisticated as that implementation.